### ISSUE TITLE
Should default deployment clean out directories?

### ISSUE THREAD

**Issue by hadley** [r-lib/pkgdown#1394]
To avoid problems like https://github.com/tidyverse/dplyr/issues/5476

**Comment by maelle** [comment]
@hadley https://github.com/r-lib/pkgdown/pull/1629#discussion_r619341143 was a reminder of this

Now, if one wanted to add arbitrary pages/files to the directory, how would one go about that
* add custom pages back after the website is built (so not possible to use default deployment)
* for files, they would be in assets

**Comment by maelle** [comment]
I think I had overthought this, the idea is simply to (breaking) change the default value of clean to TRUE?

**Comment by hadley** [comment]
Yeah, I think so. Maybe we can make this conditional on using the latest template? OTOH, I'm not sure whether this is likely to break any existing behaviour so it might be fine to change unconditionally.
